src/common/nft.extendedattributes.service.ts

```diff
@@ -0,0 +1,74 @@
+import { Injectable } from "@nestjs/common";
+import { NftMetadata } from "src/endpoints/tokens/entities/nft.metadata";
+import { ApiUtils } from "src/utils/api.utils";
+import { BinaryUtils } from "src/utils/binary.utils";
+import { Constants } from "src/utils/constants";
+import { ApiConfigService } from "./api.config.service";
+import { ApiService } from "./api.service";
+import { CachingService } from "./caching.service";
+
+@Injectable()
+export class NftExtendedAttributesService {
+
+  constructor(
+    private readonly cachingService: CachingService,
+    private readonly apiConfigService: ApiConfigService,
+    private readonly apiService: ApiService,
+  ) {}
+
+  async getExtendedAttributesFromRawAttributes(attributes: string): Promise<NftMetadata | undefined> {
+    let metadata = this.getMetadata(attributes);
+    if (metadata === undefined) {
+      return undefined;
+    }
+
+    return this.getExtendedAttributesFromMetadata(metadata);
+  }
+
+  async getExtendedAttributesFromMetadata(metadata: string): Promise<NftMetadata | undefined> {
+    let result = await this.cachingService.getOrSetCache<NftMetadata>(
+      `nftExtendedAttributes:${metadata}`,
+      async () => await this.getExtendedAttributesFromIpfs(metadata ?? ''),
+      Constants.oneWeek(),
+      Constants.oneDay()
+    );
+
+    if (Object.keys(result).length > 0) {
+      if (result.fileUri) {
+        result.fileUri = ApiUtils.replaceUri(result.fileUri, 'https://ipfs.io/ipfs', this.apiConfigService.getMediaUrl() + '/nfts/asset');
```

Comment:
we need a common function that processes this url, since it is used in other places as well